app/src/include/firebase/future.h

```diff
@@ -20,6 +20,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <mutex>
```

Comment:
Can you use firebase::Mutex and firebase::MutexLock instead?

Reply:
As discussed, this change will be made in a follow-up PR. We want to get this merged to fix the flake in CI.

Reply:
After discussion, we're going to switch this to firebase::Mutex / firebase::MutexLock in a follow-up PR, as it also requires moving mutex.h to app/src/include/firebase/internal (and adding #ifndef DOXYGEN to avoid documenting the internal class).